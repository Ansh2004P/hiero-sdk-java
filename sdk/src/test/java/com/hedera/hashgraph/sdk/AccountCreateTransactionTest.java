package com.hedera.hashgraph.sdk;

import org.junit.AfterClass;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.threeten.bp.Instant;

import static io.github.jsonSnapshot.SnapshotMatcher.*;

public class AccountCreateTransactionTest {
    private static final PrivateKey privateKey = PrivateKey.fromString(
        "302e020100300506032b657004220420db484b828e64b2d8f12ce3c0a0e93a0b8cce7af1bb8f39c97732394482538e10");

    final Instant instant = Instant.ofEpochSecond(1554158542);

    @BeforeAll
    public static void beforeAll() {
        start();
    }

    @AfterClass
    public static void afterAll() {
        validateSnapshots();
    }

    Transaction setTransaction() {
        var client = Client.forTestnet();
        return new AccountCreateTransaction()
            .setNodeAccountId(AccountId.fromString("0.0.5005"))
            .setTransactionId(new TransactionId(AccountId.fromString("0.0.5006"), instant))
            .setKey(privateKey.getPublicKey())
            .setInitialBalance(Hbar.fromTinybar(450))
            .setProxyAccountId(AccountId.fromString("0.0.1001"))
            .setReceiverSignatureRequired(true)
            .setMaxTransactionFee(Hbar.fromTinybar(100_000))
            .build(client)
            .sign(privateKey);
    }

    @Test
    @DisplayName("object to be sent matches snapshot")
    void matchesSnap() {
        expect(setTransaction().makeRequest()).toMatchSnapshot();
    }
}
