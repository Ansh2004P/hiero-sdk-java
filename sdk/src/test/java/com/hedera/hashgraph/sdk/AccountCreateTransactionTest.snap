com.hedera.hashgraph.sdk.AccountCreateTransactionTest.shouldSerialize=[
  "\nsig_map {\n  sig_pair {\n    ed25519: \"7\\321\\034\\342\\300\\021VL\\333\\303\\344\\305\\310K\\333\\005\\340\\177\\374\\337^#\\v>M\\'\\331\\304\\321\\301\\236z\\365O\\355\\3163\\202\\036\\227L~\\251\\217U8YX=\\034\\305\\235}\\354&/RqE\\215\\221(\\336\\b\"\n    pub_key_prefix: \"\\340\\310\\354\\'X\\245\\207\\237\\372\\302&\\241<\\fQky\\236r\\343QA\\240\\335\\202\\217\\224\\323y\\210\\244\\267\"\n  }\n}\n\ncrypto_create_account {\n  auto_renew_period {\n    seconds: 7776000\n  }\n  initial_balance: 450\n  key {\n    ed25519: \"\\340\\310\\354\\'X\\245\\207\\237\\372\\302&\\241<\\fQky\\236r\\343QA\\240\\335\\202\\217\\224\\323y\\210\\244\\267\"\n  }\n  proxy_account_i_d {\n    account_num: 1001\n    realm_num: 0\n    shard_num: 0\n  }\n  receive_record_threshold: 9223372036854775807\n  receiver_sig_required: true\n  send_record_threshold: 9223372036854775807\n}\nnode_account_i_d {\n  account_num: 5005\n  realm_num: 0\n  shard_num: 0\n}\ntransaction_fee: 100000\ntransaction_i_d {\n  account_i_d {\n    account_num: 5006\n    realm_num: 0\n    shard_num: 0\n  }\n  transaction_valid_start {\n    seconds: 1554158542\n  }\n}\ntransaction_valid_duration {\n  seconds: 120\n}"
]